src/System.Net.Http.WinHttpHandler/tests/UnitTests/WinHttpHandlerTest.cs

```diff
@@ -31,11 +32,33 @@ public void Dispose()
         }
 
         [Fact]
-        public void AutomaticRedirection_CtorAndGet_DefaultValueIsTrue()
+        public void Ctor_ExpectedDefaultPropertyValues()
         {
             var handler = new WinHttpHandler();
-            
-            Assert.True(handler.AutomaticRedirection);
+
+            Assert.Equal(SslProtocols.Tls | SslProtocols.Tls11 | SslProtocols.Tls12, handler.SslProtocols);
+            Assert.Equal(true, handler.AutomaticRedirection);
+            Assert.Equal(50, handler.MaxAutomaticRedirections);
+            Assert.Equal(DecompressionMethods.Deflate | DecompressionMethods.GZip, handler.AutomaticDecompression);
+            Assert.Equal(CookieUsePolicy.UseInternalCookieStoreOnly, handler.CookieUsePolicy);
+            Assert.Equal(null, handler.CookieContainer);
+            Assert.Equal(null, handler.ServerCertificateValidationCallback);
+            Assert.Equal(false, handler.CheckCertificateRevocationList);
+            Assert.Equal(ClientCertificateOption.Manual, handler.ClientCertificateOption);
+            X509Certificate2Collection certs = handler.ClientCertificates;
+            Assert.True(certs.Count == 0);
+            Assert.Equal(false, handler.PreAuthenticate);
+            Assert.Equal(null, handler.ServerCredentials);
+            Assert.Equal(WindowsProxyUsePolicy.UseWinHttpProxy, handler.WindowsProxyUsePolicy);
+            Assert.Equal(CredentialCache.DefaultCredentials, handler.DefaultProxyCredentials);
+            Assert.Equal(null, handler.Proxy);
+            Assert.Equal(Int32.MaxValue, handler.MaxConnectionsPerServer);
```

Comment:
Opened #3729 to discuss the default: my suggestion is to get a number maybe >2 but definitely not allow an infinite number of connections by default.